packages/std/src/query.rs

```diff
@@ -61,3 +63,66 @@ pub struct AllBalanceResponse {
     // Returns all non-zero coins held by this account.
     pub amount: Vec<Coin>,
 }
+
+#[cfg(feature = "staking")]
+pub use staking::{Delegation, DelegationsResponse, StakingQuery, Validator, ValidatorsResponse};
+
+#[cfg(feature = "staking")]
+mod staking {
+    use schemars::JsonSchema;
+    use serde::{Deserialize, Serialize};
+
+    use crate::coins::Coin;
+    use crate::types::HumanAddr;
+
+    #[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+    #[serde(rename_all = "snake_case")]
+    pub enum StakingQuery {
+        Validators {},
+        // Delegations will return all delegations by the delegator,
+        // or just those to the given validator (if set)
+        Delegations {
+            delegator: HumanAddr,
+            validator: Option<HumanAddr>,
+        },
+    }
+
+    #[cfg(feature = "staking")]
+    #[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+    /// ValidatorsResponse is data format returned from StakingRequest::Validators query
+    pub struct ValidatorsResponse {
+        pub validators: Vec<Validator>,
+    }
+
+    #[cfg(feature = "staking")]
+    #[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+    pub struct Validator {
+        pub address: HumanAddr,
+        // rates are denominated in 10^-6 - 1_000_000 (max) = 100%, 10_000 = 1%
+        // TODO: capture this in some Dec type?
+        pub commission: u64,
+        pub max_commission: u64,
+        // what units are these (in terms of time)?
+        pub max_change_rate: u64,
+    }
+
+    #[cfg(feature = "staking")]
+    #[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+    #[serde(rename_all = "snake_case")]
+    /// DelegationsResponse is data format returned from StakingRequest::Delegations query
```

Comment:
Side note: Rust will most likely converge towards doc comments first. The reasoning is to better support long doc comments, that otherwise move other attributes out of the screen. This is specified but not yet implemented in rustfmt: https://github.com/rust-lang/rustfmt/issues/3303

No issue here as tooling will sort them for us one day and we don't typically have long doc comments.

Reply:
Nice note, I have no opinion, put them close to the struct to make sure they bind, but the other way (doc comments first) does look cleaner.